### PR TITLE
Decouple version from app.yaml/Dockerfile in flex deploy

### LINKED ISSUE #639: Make deployment version part of main configuration
The user should be able to easily override the default version configuration without having to specify app.yaml and Dockerfile. Version configuration should also be given more prominence by moving it outside of the "Custom Configuration" panel.

deployment_config

### PR DESCRIPTION
**Pull request by etanshaul**
fixes #639 

image

### CODE REVIEW

**Comment by etanshaul** [comment]
@patflynn, @meltsufin

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ public AppEngineDeployAction createAutoDeploymentAction(
         artifactToDeploy,
         defaultAppYaml(),
         defaultDockerfile(artifactType),
-        AppEngineUtil.generateVersion(),
+        version,
```
Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java]: how does the user generate the version now?=

Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java]: If the user doesn't provide a version ID by typing one into the textfield then one will be generated automatically (see `AppEngineDeployAction` line 113).

Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java]: ah interesting.  I'm not crazy about that duplication.  How about we just make the versionId param to AppEngineDeployAction @NotNull?

Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java]: The duplication between `createCustomDeploymentAction` and `createAutoDeploymentAction`?

Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java]: No, The fact that we are generating the version here, but then the DeployAction actually calls generate version itself if no version is passed.  I'd rather have that logic confined to the helper.

Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java]: The only place the version is being generated is in DeployAction. Regardless, you are right its better to confine it to the helper. I'll update it.

**Comment by etanshaul** [comment]
refactored version generation per conversion. @patflynn let me know if that's what you had in mind.

**Comment by patflynn** [comment]
LGTM

**Pull request merged**